numba/core/extending.py

```diff
@@ -352,6 +353,19 @@ def _uuid(self):
             self._set_uuid(u)
         return u
 
+    if PYVERSION in ((3, 14), ):
+        # In 3.14 __annotations__ is a function, not a dictionary. So it is not
+        # longer updated using functools, but must be added explicitly.
+        #
+        # See also: https://peps.python.org/pep-0649/
+        @property
+        def __annotations__(self):
+            return self._defn.__annotations__
```

Comment:
@esc, do you recall if the lack of this method causing any test to fail? `typing.get_type_hints` work expectedly when I commented this method out.

Reply:
According to PEP 649, `__annotations__` may not exist. In the test, we should use `inspect.get_annotations(void_func)` instead of `void_func.__annotations__`. Then we can remove the need for defining `__annotations__` in `_Intrinsic`.

Reply:
I have reverted 8915e2c at: 1fbada8c36635d080f3fa8443b2f0b45a0d1e257 and added the suggested fix at: ac7bd1b6164eb99b367f3526da7227a991bee056 -- the test `numba.tests.test_extending.TestIntrinsic.test_docstring` passes for me locally with this alternative fix.